pkg/util/compat.go

```diff
@@ -4,11 +4,13 @@ import (
 	v1 "k8s.io/api/core/v1"
 )
 
+// See https://kubernetes.io/docs/reference/labels-annotations-taints/
+
 func GetRegion(labels map[string]string) (string, bool) {
-	if _, ok := labels[v1.LabelZoneRegion]; ok {
+	if _, ok := labels[v1.LabelTopologyRegion]; ok { // Label as of 1.17
+		return labels[v1.LabelTopologyRegion], true
+	} else if _, ok := labels[v1.LabelZoneRegion]; ok { // deprecated label
```

Comment:
Was this, specifically, tested? If so, just done manually on one of our GCP clusters?

Reply:
So, this is used here: https://github.com/kubecost/cost-model/blob/mmd-update-k8s-libraries/pkg/costmodel/costmodel.go#L977

This result is, eventually, consumed by the emitter of `kubecost_node_is_spot` (among other things) which includes region info. I am seeing that accurate on a GKE cluster (showing `us-east1` for a cluster running in `us-east1-b`).

Reply:
This matches up with pre-change behavior, as far as I can tell with a fresh cluster and the regular helm chart.

Reply:
It's also used in the providers, including GCP provider: https://github.com/kubecost/cost-model/blob/mmd-update-k8s-libraries/pkg/cloud/gcpprovider.go#L1180 and I'm not seeing related errors in the logs.

Reply:
Sounds good